analyzers/src/SonarAnalyzer.CFG/ShimLayer/IOperationWrapperSonar.cs

```diff
@@ -24,37 +24,31 @@
 namespace StyleCop.Analyzers.Lightup
 {
     // This is a temporary substitute for IOperationWrapper in case StyleCop will accept PR https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3381
-    public class IOperationWrapperSonar
+    public readonly struct IOperationWrapperSonar
```

Comment:
@martin-strecker-sonarsource can we use https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/StyleCop.Analyzers/StyleCop.Analyzers.CodeGeneration/OperationLightupGenerator.cs#L21 instead?

see https://github.com/DotNetAnalyzers/StyleCopAnalyzers/commit/9b5db190f53bc6e8689b4df433684d92868fb6b0

Reply:
This would require an update to the SourceGenerator as well and will break a lot of existing code (there are several PRs in GH related to this change with a lot of files changed). We can and should absolutely do this. But it isn't an easy effort. We can however still take my solution (which by the way was implemented in StyleCop almost the same way before they moved it it to the source generator: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/f8f0b4b126f55c4619d0930c4a5aeb59f0c572a2/StyleCop.Analyzers/StyleCop.Analyzers/Lightup/IOperationWrapper.cs) as a quick fix for the allocation issue.